src/coreclr/src/jit/instr.cpp

```diff
@@ -648,118 +648,39 @@ void CodeGen::inst_TT(instruction ins, GenTree* tree, unsigned offs, int shfv, e
     }
 }
 
-/*****************************************************************************
- *
- *  Generate an instruction that has one operand given by a tree (which has
- *  been made addressable) and another that is a register.
- */
-
-void CodeGen::inst_TT_RV(instruction ins, GenTree* tree, regNumber reg, unsigned offs, emitAttr size, insFlags flags)
+//------------------------------------------------------------------------
+// inst_TT_RV: Generate a store of a lclVar
+//
+// Arguments:
+//    ins  - the instruction to generate
+//    tree - the lclVar node
+//    reg  - the register currently holding the value of the local
+//    size - the size attributes for the store
+//
+void CodeGen::inst_TT_RV(instruction ins, GenTree* tree, regNumber reg, emitAttr size)
 {
+    // The tree must have a valid register value.
     assert(reg != REG_STK);
-
-AGAIN:
-
-    /* Is this a spilled value? */
-
-    if (tree->gtFlags & GTF_SPILLED)
-    {
-        assert(!"ISSUE: If this can happen, we need to generate 'ins [ebp+spill]'");
-    }
-
-    if (size == EA_UNKNOWN)
+    bool isValidInReg = ((tree->gtFlags & GTF_SPILLED) == 0);
```

Comment:
`isValidInReg` is used only in an assert, it would be good to put all this code in an `ifdef DEBUG`.